api/src/main/java/org/jboss/marshalling/util/IdentityIntMap.java

```diff
@@ -18,17 +18,35 @@
 
 package org.jboss.marshalling.util;
 
+import java.security.AccessController;
+import java.security.PrivilegedAction;
 import java.util.Arrays;
 
 /**
  * An efficient identity object map whose keys are objects and whose values are {@code int}s.
  */
 public final class IdentityIntMap<T> implements Cloneable {
+
+    private static final String THRESHOLD_PROPERTY_NAME = "jboss.marshalling.identityint.reinitialize.capacity.threshold";
+    private static final int DEFAULT_REINITIALIZE_CAPACITY_THRESHOLD = 1 << 16; // 65536
+    private static final int REINITIALIZE_CAPACITY_THRESHOLD;
+
     private int[] values;
     private Object[] keys;
     private int count;
     private int resizeCount;
 
+    private final int initialCapacity;
+    private final float loadFactor;
+
+    static {
+        if (System.getSecurityManager() == null) {
+            REINITIALIZE_CAPACITY_THRESHOLD = Integer.getInteger(THRESHOLD_PROPERTY_NAME, DEFAULT_REINITIALIZE_CAPACITY_THRESHOLD);
+        } else {
+            REINITIALIZE_CAPACITY_THRESHOLD = AccessController.doPrivileged((PrivilegedAction<Integer>) () -> Integer.getInteger(THRESHOLD_PROPERTY_NAME, DEFAULT_REINITIALIZE_CAPACITY_THRESHOLD));
+        }
```

Comment:
This is a static initializer; it is run only once.  There's no useful benefit to having two branches here: better to just always use a privileged block.

Reply:
You are correct. I thought that it's better to execute without a privileged block if the security manager is not enabled. However, as you pointed out, it's run only once, so this can be simplified.